Client/FluentClientExtensions.cs

```diff
@@ -21,8 +21,8 @@ public static class FluentClientExtensions
         public static bool Remove<TFilter>(this List<IHttpFilter> filters)
             where TFilter : IHttpFilter
         {
-            TFilter filter = filters.OfType<TFilter>().FirstOrDefault();
-            return filter != null && filters.Remove(filter);
+            IEnumerable<TFilter> tFilters = filters.OfType<TFilter>();
+            return tFilters.Any() && filters.Remove(tFilters.First());
```

Comment:
@Jericho This is a bit less efficient since it searches the filters twice. In retrospect, I'm not sure stopping at the first match is intuitive — if I ran `client.Filters.Remove<IFilter>()`, I'd expect it to remove all matches.

How about something like this instead?
```c#
return filters.RemoveAll(p => p is TFilter) > 0;
```